doc/user_guide/io.rst

```diff
@@ -1030,6 +1032,162 @@ HDF5 file. All other keyword arguments will be passed to
+    /entry/xsp3_addetector
+    /entry/xsp3_addetector_total
+    HDF datasets found
+    /entry/solstice_scan/keys/uniqueKeys
+    /entry/solstice_scan/scan_shape
+    Out[3]:
+    (['/entry/xsp3_addetector', '/entry/xsp3_addetector_total'],
+     ['/entry/solstice_scan/keys/uniqueKeys', '/entry/solstice_scan/scan_shape'])    
+    
+
+Reading
+^^^^^^^
+Nexus files can contain multiple datasets within the same file.
+HyperSpy supports reading in one or more datasets however Nexus
+can contain a vary large number of datasets.
+
+Extra loading arguments
++++++++++++++++++++++++
+- ``dset_search_keys`` : ``str`` or ``list`` 
+Absolute path(s) or substring(s) to use to find
+one or more datasets. (default is ``None`` all Nexus Datasets will be read)
+- ``meta_search_keys`` : ``str`` or ``list`` of strings 
+Absolute path(s) or substring(s) to use to find
+one or more datasets. (default is ``None`` all Nexus Metadata will be read)
+ 
+Reading the sole Nexus dataset within a Nexus file:
+
+.. code-block:: python
+
+    >>> hs.load("sample.nxs")
+
+If multiple datasets are present within the Nexus file and you try the same 
+command again **all** available NXdata datasets will be loaded.
+To load NXdata and hdf datasets as well
+
+.. code-block:: python
+
+    >>> hs.load("sample.nxs",nxdata_only=False)
+
+.. note::
+
+    Given that HDF5 files can accommodate very large datasets, setting 
+    ``lazy=True`` is strongly recommended if the contents of the HDF5 file 
+    are not known apriori. This prevents issues with regard to loading datasets
+    far larger than memory.
+
+    Also note that setting ``lazy=True`` leaves the file handle to the 
+    HDF5 file open.
+    If it is important that the files be closed after reading,
+    set ``lazy=False``.
+
+.. code-block:: python
+
+    >>> hs.load("sample.h5")
+    [<Signal2D, title: HAADF, dimensions: (|128, 128)>,
+    <Signal1D, title: EELS, dimensions: (|64, 64, 1024)>]
+
+We can load a specific datasets using the ``dset_search_path`` keyword argument. 
+Setting it to the absolute path of the desired dataset will cause 
+the single dataset to be loaded.
+Setting it to a substring or list of substring, datasets who's paths contain 
+one of the substrings will be loaded.
+
+
+.. code-block:: python
+
+    >>> # Loading a specific dataset
+    >>> hs.load("sample.h5", dset_search_path='/entry/experiment/EDS/data')
+
+We can also choose to load only specific items from the metadata using
+ ``dset_path`` keyword argument. This can be a list of 
+absolute paths of the desired dataset or substrings to find in the metadata
+path.
+
+.. code-block:: python
+
+    >>> # Loading a specific dataset
+    >>> hs.load("sample.h5", dset_search_path='EDS')
+
+Nexus files also support parameters or dimensions that have been varied 
+non-linearly.
+Since HyperSpy Signals expect linear variation of parameters / axes, such 
+non-linear information would be lost in the axes manager and replaced with
+indices
```

Comment:
@ericpre @jlaehne @francisco-dlp - how can we sort #1927 ? I spoke to people trying to use hyperspy for Raman data, CL, PL data (and diffraction data) yesterday and this functionality is really needed for all these applications...